stdlib/javascript/compile.mc

```diff
@@ -121,56 +210,104 @@ lang MExprJSCompile = MExprAst + JSProgAst
     match rec [] app with (fun, args) then
       -- Function calls
       match fun with TmVar { ident = ident } then
-        JSEApp { fun = JSEVar { id = ident }, args = map compileExpr args }
+        JSEApp { fun = JSEVar { id = ident }, args = map (compileMExpr opts) args, curried = true }
 
       -- Intrinsics
       else match fun with TmConst { val = val } then
-        let args = map compileExpr args in
-        compileOp fun args val
+        let args = map (compileMExpr opts) args in
+        compileCOp opts args val
 
-      else error "Unsupported application in compileExpr"
+      else error "Unsupported application in compileMExpr"
     else never
 
   -- Anonymous function, not allowed.
-  | TmLam _ -> error "Anonymous function in compileExpr."
+  | TmLam { ident = arg, body = body } ->
+    let body = (compileMExpr opts) body in
+    -- dprintLn "Compiling lambda";
+    -- dprintLn body;
+    JSEFun { param = arg, body = body }
 
   -- Unit type is represented by int literal 0.
   | TmRecord { bindings = bindings } ->
-    if mapIsEmpty bindings then JSEInt { i = 0 }
-    else error "ERROR: Records cannot be handled in compileExpr."
+    let fieldSeq = mapToSeq bindings in
+    let compileField = lam f. match f with (sid, expr)
+      then (sidToString sid, (compileMExpr opts) expr)
+      else never in
+    JSEObject { fields = map compileField fieldSeq }
 
-  | TmSeq {tms = tms, ty = ty, info = info} & t ->
-    -- Special handling of strings
+  | TmSeq {tms = tms} ->
     -- Check if sequence of characters, then concatenate them into a string
     if _isCharSeq tms then
       match (_charSeq2String tms) with Some str then JSEString { s = str }
```

Comment:
How come you are special casing on strings here?

Reply:
Because I want hard-coded strings to become strings in the compiled JS as well. Let's say `printLn "nothing"` should become `printLn("nothing")` and not `printLn(['n', 'o', 't', 'h', 'i', 'n', 'g'])`.